testing/kuttl/e2e/password-change/README.md

```diff
@@ -0,0 +1,25 @@
+### Password Change Test with Kuttl
+
+This Kuttl routine runs through the following steps:
+
+#### Create cluster and test connection
+
+- 00: Creates the cluster and verifies that it exists and is ready for connection
+- 01: Connects to the cluster with the PGO-generated password (both with env vars and with the URI)
+
+#### Default user connection tests
+
+- 02: Change the password (using Kuttl's update object method on the secret's `data` field) and verify that the password changes
+- 03: Connects to the cluster with the user-defined password (both with env vars and with the URI)
+- 04: Change the password (using Kuttl's update object method on the secret's `stringData` field) and verify that the password changes
+- 05: Connects to the cluster with the second user-defined password (both with env vars and with the URI)
```

Comment:
🤔 Is the only difference between 2+3 and 4+5 that one uses `data` and `stringData`? I'm not opposed to testing Kubernetes features but doing both of these feels redundant. If one were to go away, I think I would keep `stringData` and the assertion that it ends up in `data`.

🤔 Perhaps it would be good to test that [`verifier` can be assigned](https://access.crunchydata.com/documentation/postgres-operator/v5/architecture/user-management/#custom-passwords) near here?

Reply:
Yeah, agree there's some redundancy here that I wanted to talk about -- I _mostly_ followed the ticket, but now that you put it that way, it makes more sense to include a test where the user sets the verifier rather than two nearly identical ways to set the password.

Reply:
Updated with verifier tests